### ISSUE TITLE
Source Code?

### ISSUE THREAD

**Issue by mpathy** [ArtemGr/Sidekick#46]
Hi there.. I wanted to ask if the source code for this project is available.. I would like to add some stuff for our online pen & paper group, if possible..

You added a MIT license here, so I was optimistic and posted this issue ;)

**Comment by Demosphere** [comment]
I second this.

**Comment by izaclorimer** [comment]
I would also like this.

**Comment by kacperski1** [comment]
I'd also love to get the source code. The bot is available for free anyway, so why not let us boost the development a bit?

**Comment by FluxAugur** [comment]
+1

**Comment by mmmeff** [comment]
This is sketch, people should not add this to their Discord unless the source is available

**Comment by yukihyou** [comment]
It's not so much an issue with sane server settings. But anyone that gives `Administrator` to anyone that isn't known to them personally is just asking for trouble, and a lot of people just assume that bots need it, and give it to them regardless.

I think that if you use `&permissions=19456` as the mask (giving only `Read Messages` `Send Messages` and `Embed Links` access) by default, this will stop people from just adding it to their stupidly-insecure "Bots" role that has Admin.

**Comment by ArtemGr** [comment]
@yukihyou Have a doc at hand? I might update the README.

Also, everyone, please share your experience with bot using tighter permissions if you do.

**Comment by yukihyou** [comment]
The actual bitmask is shown here: https://discordapp.com/developers/docs/topics/permissions
There's a handy dandy calculator here: https://discordapi.com/permissions.html

Note that this bot is really simple (no moderator commands, voice chat, music, etc) so all it really needs is Read and Send permissions in whatever channels you want to use it in. Which, given that most users have that already, shouldn't even need a role most of the time. The only problem with giving `&permissions=0" is that it doesn't actually create itself a limited role - it needs at least one permission set to create itself a restricted role.

Unfortunately, a lot of bots are going down the "just tick admin then you don't have to worry about stuff not working" route, which is training server admins to just create a "Bots" role with Administrator and assign it to any bot they bring in, because then it "just works" ... The concept of "least privilege" is something that should be encouraged. As an example, Sidekick is currently running in my server with a role that adds no permissions at all - it's the same as any other guest user - and it works just fine.

Also, given the "wontfix" label, I presume you're not interested in people submitting bugfixes/enhancements via pull request?

**Comment by ArtemGr** [comment]
Thanks for the URLs, @yukihyou! I'll check it out.

> Also, given the "wontfix" label, I presume you're not interested in people submitting bugfixes/enhancements via pull request?

Feel free to create a separate issue for permissions or to submit a pull request with a README patch.

**Comment by yukihyou** [comment]
I meant, with regard to sharing the source in general (this issue ticket) - letting us bored coders add/fix things that are sitting in your backlog. :)

**Comment by ArtemGr** [comment]
If you can write Rust and have an issue in mind, then by all means, mention that in the corresponding issue and we'll think of something.

**Comment by ifohancroft** [comment]
I would like the source code as well. I mean, come on, it is issued under the MIT license. If I am not mistaken and reading something wrong, you are actually breaking the license by not giving us the code. Also, the fact that you don't even address us, wanting the source code is just plain rude.

**Comment by klaude** [comment]
> If you can write Rust and have an issue in mind

That's kind of the point of using an open source site like this, @ArtemGr. If you want to use GitHub for issue tracking then remove the project's license file and say that this is here for issue tracking in the readme. Others are right that this violates the MIT license. Otherwise, throw the source up! Whatcha got to lose?

**Comment by ifohancroft** [comment]
@klaude Apparently he has a patreon page and its sole purpose is for people to support him via money for the bot/if they like it/use it/etc so I guess that's why he refuses to give the source.

**Comment by klaude** [comment]
Fair enough, @ifohancroft. We've all gotta eat. All he has to do is take the license down and say up front what the project's GitHub presence is for and all is well. At least for me. Otherwise I wouldn't mind seeing the source.

**Comment by ifohancroft** [comment]
@klaude same here.

**Comment by ifohancroft** [comment]
I just wanted to say that I am officially stopping to use the bot because of the creator's violation of the license and his refusal to address that. I will right now remove the bot from my server and I urge everyone to do the same.

**Comment by ArtemGr** [comment]
The MIT license is here to clarify the status of published source code, such as this one: https://github.com/ArtemGr/Sidekick/issues/90#issuecomment-351323857.
